From 728824d1a2f964f91d4b3cf092cdf3d4d983c162 Mon Sep 17 00:00:00 2001
From: Victor Gu <xigu@marvell.com>
Date: Mon, 24 Apr 2017 16:27:55 +0800
Subject: [PATCH 0967/1345] dts: a37xx: add reserved memory for PSCI area

commit  e001f73e45596eae33778111aaac6834a78fe4ba from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

The PSCI area should be reserved in Linux for PSCI operations.
This patch reserves 2M byte memory for PSCI, which covers all PSCI
code and data area, and be 2M byte aligned which is required by
Linux 2M byte big pages.

Change-Id: I40eb03bda6e7897512acfe76163e99b1309ca0fb
Signed-off-by: Victor Gu <xigu@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/38839
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Kostya Porotchkin <kostap@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-37xx.dtsi |   11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
index 257158a..9c060e7 100644
--- a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
@@ -72,6 +72,17 @@
 		};
 	};
 
+	reserved-memory {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
+		psci-area@4000000 {
+			reg = <0 0x4000000 0 0x200000>;
+			no-map;
+		};
+	};
+
 	cpus {
 		#address-cells = <1>;
 		#size-cells = <0>;
-- 
1.7.9.5

